File: server/models/category.go
// 品类模型（映射 categories 表）
package models

import "time"

// Category 品类结构
type Category struct {
	Id        int64     `orm:"column(id);auto" json:"id"`
	Name      string    `orm:"column(name);size(64)" json:"name"`
	ParentId  int64     `orm:"column(parent_id)" json:"parent_id"`
	Icon      string    `orm:"column(icon);size(255)" json:"icon"`
	Sort      int       `orm:"column(sort)" json:"sort"`
	Status    int       `orm:"column(status)" json:"status"`
	CreatedAt time.Time `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
}

// TableName 指定表名
func (c *Category) TableName() string {
	return "categories"
}

File: apps/flutter/lib/models/category.dart
/// 类目,字段对齐 server/models/category.go。
class Category {
  final int id;
  final String name;
  final String icon;
  Category({required this.id, required this.name, required this.icon});

  factory Category.fromJson(Map<String, dynamic> j) => Category(
        id: j['id'] as int? ?? 0,
        name: j['name'] as String? ?? '',
        icon: j['icon'] as String? ?? '',
      );
}
